app/scripts/modules/core/src/plugins/deck.plugin.ts

```diff
@@ -21,6 +24,7 @@ export interface IDeckPlugin {
 export function registerPluginExtensions(plugin: IDeckPlugin): PromiseLike<any> {
   plugin.stages?.forEach(stage => Registry.pipeline.registerStage(stage));
   plugin.preconfiguredJobStages?.forEach(stage => Registry.pipeline.registerPreconfiguredJobStage(stage));
+  plugin.resourceKinds?.forEach(kind => registerResourceKind(kind));
```

Comment:
@vigneshm I _kinda_ think I _maybe_ figured out how to expose this to plugins via the `pluginsdk` contract? Maybe? LMK if you think this will work or if I went down my own special path 😆

Reply:
Yes, this is the way to open it up!

Reply:
woohoo! thanks vignesh :D